fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/ImportHandlerUtils.java

```diff
@@ -53,10 +53,17 @@ public static Integer getNumberOfRows(Sheet sheet, int primaryColumn) {
         Integer noOfEntries = 0;
         // getLastRowNum and getPhysicalNumberOfRows showing false values
         // sometimes
-        while (sheet.getRow(noOfEntries + 1) != null && sheet.getRow(noOfEntries + 1).getCell(primaryColumn) != null) {
+        while (true) {
```

Comment:
Please rewrite to this:
```
int maxRows = sheet.getLastRowNum(); // Or set a hard limit, e.g., 10_000

while (noOfEntries < maxRows) {
    Row row = sheet.getRow(noOfEntries + 1);
    if (row == null) {
        break;
    }
    Cell cell = row.getCell(primaryColumn);
    if (cell == null || cell.getCellType() == CellType.BLANK) {
        break;
    }
    noOfEntries++;
}
```

This a little bit better as it never an endless loop